sdks/python/apache_beam/runners/direct/watermark_manager.py

```diff
@@ -211,12 +211,9 @@ def refresh(self):
       min_pending_timestamp = WatermarkManager.WATERMARK_POS_INF
       has_pending_elements = False
       for input_bundle in self._pending:
-        # TODO(ccy): we can have the Bundle class keep track of the minimum
-        # timestamp so we don't have to do an iteration here.
-        for wv in input_bundle.get_elements_iterable():
-          has_pending_elements = True
-          if wv.timestamp < min_pending_timestamp:
-            min_pending_timestamp = wv.timestamp
+        if input_bundle.has_elements():
+            min_pending_timestamp = min(min_pending_timestamp, input_bundle.get_min_timestamp())
```

Comment:
<!--new_thread; commit:61ae92dcffa98203cbe8f56930802405de2524f0; resolved:0-->
Please use 2sp indentation, wrap line to 80 chars.